doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+# KUKSA requirements
+
+
+# Content
+- [KUKSA requirements](#kuksa-requirements)
+- [Content](#content)
+- [Documentation](#documentation)
+    - [Terminology](#terminology)
+    - [System Architecture](#system-architecture)
+- [Functional requirements](#functional-requirements)
+    - [As Client](#as-client)
+    - [As Databroker](#as-databroker)
+    - [As Provider](#as-provider)
+- [Non-Functional Requirements](#non-functional-requirements)
+  - [Data availability/persistence according to lifecycle of Client, Databroker and Provider](#data-availabilitypersistence-according-to-lifecycle-of-client-databroker-and-provider)
+  - [Wildcard support](#wildcard-support)
+  - [Registration of Datapoints](#registration-of-datapoints)
+  - [Availability of Datapoints](#availability-of-datapoints)
+  - [Lifecycle of components](#lifecycle-of-components)
+  - [Path requests](#path-requests)
+  - [Errors](#errors)
+  - [Setting Values](#setting-values)
+  - [Atomic operatoins](#atomic-operatoins)
+  - [Update notifications](#update-notifications)
+  - [Access rights](#access-rights)
+  - [VSS signals - Users vs providers](#vss-signals---users-vs-providers)
```

Comment:
To me the whole document more reads like a mix between requirements, design considerations, design decisions and and API definition. Maybe it becomes more accessible and when we split the document into different documents which can be discussed separately. Otherwise, I think it will be very different and lengthy to achieve consensus on the whole PR. 

After a first read, this the part before the discussion of the `VSS signals - Users vs providers` could be in another document than what follows afterwards.

Reply:
My idea was/is to have a document containing both design considerations together with the resulting requirements. The requirements could be tagged, so that they can be filtered from the rest later.

Reply:
I See, and linking the design considerations and the requirements sounds good. Given the overall length of the document, I would still opt to keep it on multiple pages and use cross-references. 
In the long run, I would hope that the documents that we get out of this process can be the foundation for creating something like a Kuksa API specification in a similar style as other specs like VISSv2 (https://github.com/COVESA/vehicle-information-service-specification) 

Maybe more of a philosophical question, but should the requirements be more the result of the use cases and user expectations, and then the design considerations be the answer on how to address the requirements?

Reply:
That's a thing which I can still not fully grab in my head. I was thinking about having a few "basic requirements" what consumers (and providers) can expect from the broker. I.e. things like "I want to pull/get the current state of vehicle signals via a single service interface (in an efficient way)". This does not yet describe certain details. 

Based on that set of requirements some "design considerations" need to be discussed and decided, resulting in a more detailed "next level" set of requirements addressed to towards the API design and broker implementation.